service/dp3.go

```diff
@@ -60,11 +59,10 @@ func (dp3 *DP3) Start(ctx context.Context, options ...DP3Option) error { //nolin
 		return fmt.Errorf("failed to ping database at %s: %w", dbpath, err)
 	}
 	ns := nodestore.NewNodestore(store, cache)
-	rm, err := rootmap.NewSQLRootmap(db)
+	rm, err := rootmap.NewSQLRootmap(ctx, db, 1e9)
```

Comment:
let's adopt a functional options pattern for the rootmap, similar to what we have for the walmgr https://github.com/wkalt/dp3/blob/main/wal/options.go#L20,L62

that will be a little easier to read than the bare 1e9